### PR TITLE
Switch the field to an Item

### LINKED ISSUE #56: Shouldn't parameters be semicolon-separated?
Over in https://mailarchive.ietf.org/arch/msg/httpapi/GRGrz_dh4ccwB696uYR21_6brU8/ I noted that it's odd that parameters are separated by commas, because in other contexts this is semantically the same as the same header repeated multiple times.

It doesn't make sense that

```
Problem: type="https://example.net/problems/almost-out"
Problem: title="you're almost out of credit"
Problem: credit_left=20
```

should be the same as

```
Problem: type="https://example.net/problems/almost-out", title="you're almost out of credit", credit_left=20
```

... This is a problem because if an intermediary appends a `Problem` header to a response that already contains a `Problem` header; the problem in the resulting message would become mangled.

I suggested using the syntax of the `Link` header, which is already well established and could easily be re-used, something like:

```
Problem: <https://example.net/problems/almost-out>; title="you're almost out of credit”; credit_left=20
```

This syntax has an [exact representation as regular expression](https://mailarchive.ietf.org/arch/msg/httpapi/Heu3hsHCNs4mkeNIgEcWCNy1CBo/).

Or `sf-item`:

```
Problem: "https://example.net/problems/almost-out"; title="you're almost out of credit”; credit_left=20
```

I don't think I got much of a formal response so I'd like to ask here.

### PR DESCRIPTION
**Pull request by mnot**
Fixes #56

### CODE REVIEW

**Review by asbjornu — APPROVED** [review]

File: draft-ietf-httpapi-rfc7807bis.md [hunk 1]
```diff
@@ -298,17 +295,19 @@ detail:
 instance:
 : The instance value (see {{instance}}), as a String
 
-If an extension member (see {{extension}}) occurs in the Problem field, its name MUST be compatible with the syntax of Dictionary keys (see {Section 3.2 of STRUCTURED-FIELDS}}) and the defining problem type MUST specify a Structured Type to serialize the value into.
+Extension members (see {{extension}} can be serialised as additional Parameters, provided that their names are compatible with the syntax of Parameter keys (see {Section 3.1.2 of STRUCTURED-FIELDS}}) and the defining problem type specifies a Structured Type to serialize the value into.
```
Comment (malevy) [inline draft-ietf-httpapi-rfc7807bis.md:298]: extra decimal point in the section reference?

File: draft-ietf-httpapi-rfc7807bis.md [hunk 2]
```diff
@@ -281,10 +281,7 @@ Problem type authors that wish their extensions to be usable in the Problem HTTP
 
 Some problems might best be conveyed in an HTTP header or trailer field, rather than in the message content. For example, when a problem does not prevent a successful response from being generated, or when the problem's details are useful to software that does not inspect the response content.
 
-The Problem HTTP field allows a limited expression of a problem object in HTTP headers or trailers. It is a Dictionary Structured Field ({{Section 3.2 of STRUCTURED-FIELDS}}) that can contain the following keys, whose semantics and related requirements are inherited from problem objects:
-
-type:
-: the type value (see {{type}}), as a String
+The Problem HTTP field allows a limited expression of a problem object in HTTP headers or trailers. It is an Item Structured Field whose value is a String ({{Section 3..3.3 of STRUCTURED-FIELDS}}) that conveys the type value (see {{type}}). It has the following optional Parameters:
```
Comment (malevy) [inline draft-ietf-httpapi-rfc7807bis.md:284]: Extra decimal point in the section reference?

Comment (mnot) [inline draft-ietf-httpapi-rfc7807bis.md:284]:
```suggestion
The Problem HTTP field allows a limited expression of a problem object in HTTP headers or trailers. It is an Item Structured Field whose value is a String ({{Section 3.3.3 of STRUCTURED-FIELDS}}) that conveys the type value (see {{type}}). It has the following optional Parameters:
```

**Review by darrelmiller — APPROVED** [review]

**Comment by mnot** [comment]
OBE

**Pull request closed without merging**